core/templates/pages/story-editor-page/modal-templates/story-editor-unpublish-modal.component.spec.ts

```diff
@@ -63,4 +79,26 @@ describe('Story Editor Unpublish Modal Component', () => {
     component.confirm();
     expect(confirmSpy).toHaveBeenCalled();
   });
+
+  it('should get status of Serial Chapter Launch Feature flag', () => {
+    expect(component.isSerialChapterFeatureFlagEnabled()).toEqual(false);
+
+    mockPlatformFeatureService.
+      status.SerialChapterLaunchCurriculumAdminView.isEnabled = true;
+    expect(component.isSerialChapterFeatureFlagEnabled()).toEqual(true);
+  });
+
+  it('should set unpublishing reason', () => {
+    expect(component.unpublishingReason).toBe('BAD_CONTENT');
+    expect(component.selectedReasonText).toBe(component.badContentReasonText);
+
+    component.setReason('CHAPTER_NEEDS_SPLITTING');
+    expect(component.unpublishingReason).toBe('CHAPTER_NEEDS_SPLITTING');
+    expect(component.selectedReasonText).toBe(
+      component.splitChapterReasonText);
+
+    component.setReason('BAD_CONTENT');
+    expect(component.unpublishingReason).toBe('BAD_CONTENT');
+    expect(component.selectedReasonText).toBe(component.badContentReasonText);
+  });
```

Comment:
```suggestion
// checks if the default reason for publishing is bad content
  it('should ...', () => {
    expect(component.unpublishingReason).toBe('BAD_CONTENT');
    expect(component.selectedReasonText).toBe(component.badContentReasonText);
 }
  it('should set unpublishing reason', () => {
    expect(component.unpublishingReason).toBe('BAD_CONTENT');
    expect(component.selectedReasonText).toBe(component.badContentReasonText);

    component.setReason('CHAPTER_NEEDS_SPLITTING');

    expect(component.unpublishingReason).toBe('CHAPTER_NEEDS_SPLITTING');
    expect(component.selectedReasonText).toBe(
      component.splitChapterReasonText);
  });
```

Reply:
Done. In the second test, I also added the test to check for setting the reason to BAD_CONTENT after CHAPTER_NEEDS_SPLITTING, else frontend coverage check is not passing.